packages/react/src/components/Comments/Comment.tsx

```diff
@@ -0,0 +1,372 @@
+   */
+  showDeleted?: boolean;
+
+  /**
+   * Whether to show reactions.
+   */
+  showReactions?: boolean;
+}
+
+/**
+ * The Comment component displays a single comment with actions,
+ * a reaction list and an editor when editing.
+ *
+ * It's generally used in the `Thread` component for comments that have already been created.
+ */
+export const Comment = ({
+  comment,
+  thread,
+  showDeleted,
+  showActions = "hover",
+  showReactions = true,
+  showResolveAction = false,
+  className,
+}: CommentProps) => {
+  const dict = useDictionary();
+
+  // TODO: review use of sub-editor
+  const commentEditor = useCreateBlockNote(
+    {
+      initialContent: comment.body,
+      trailingBlock: false,
+      dictionary: {
+        ...dict,
+        placeholders: {
+          emptyDocument: "Edit comment...",
+        },
+      },
+      schema,
+    },
+    [comment.body]
+  );
+
+  const Components = useComponentsContext()!;
+
+  const [isEditing, setEditing] = useState(false);
+
+  const editor = useBlockNoteEditor();
+
+  if (!editor.comments) {
+    throw new Error("Comments plugin not found");
+  }
+
+  const threadStore = editor.comments.threadStore;
+
+  const handleEdit = useCallback(() => {
+    setEditing(true);
+  }, []);
+
+  const onEditCancel = useCallback(() => {
+    commentEditor.replaceBlocks(commentEditor.document, comment.body);
+    setEditing(false);
+  }, [commentEditor, comment.body]);
+
+  const onEditSubmit = useCallback(
+    async (_event: MouseEvent) => {
+      // TODO: show error on failure?
+      await threadStore.updateComment({
+        commentId: comment.id,
+        comment: {
+          body: commentEditor.document,
+        },
+        threadId: thread.id,
+      });
+
+      setEditing(false);
+    },
+    [comment, thread.id, commentEditor, threadStore]
+  );
+
+  const onDelete = useCallback(async () => {
+    // TODO: show error on failure?
+    await threadStore.deleteComment({
+      commentId: comment.id,
+      threadId: thread.id,
+    });
+  }, [comment, thread.id, threadStore]);
+
+  const onReactionSelect = useCallback(
+    async (emoji: string) => {
+      // TODO: show error on failure?
+      await threadStore.addReaction({
+        threadId: thread.id,
+        commentId: comment.id,
+        emoji,
+      });
+    },
+    [comment.id, threadStore, thread.id]
+  );
+
+  const onResolve = useCallback(async () => {
+    // TODO: show error on failure?
+    await threadStore.resolveThread({
+      threadId: thread.id,
+    });
+  }, [thread.id, threadStore]);
+
+  const onReopen = useCallback(async () => {
+    // TODO: show error on failure?
+    await threadStore.unresolveThread({
+      threadId: thread.id,
+    });
+  }, [thread.id, threadStore]);
+
+  const user = useUser(editor, comment.userId);
+
+  const blockNoteContext = useBlockNoteContext();
+
+  if (!showDeleted && !comment.body) {
+    return null;
+  }
+
+  let actions: ReactNode | undefined = undefined;
```

Comment:
@matthewlipski it would be better here to extract the emoji picker component, and only call this when the user expands the picker + show a loader in between

Reply:
Turns out the `Picker` component fetches the emojis itself if a `data` prop isn't passed, which works out nicely. The response with the emojis is <80kBs, so imo it's better not to show a loader here as in most cases it'll flash for a sec before showing the picker, which is kinda jarring.